### ISSUE TITLE
DJ October 2019

### ISSUE THREAD

**Issue by xaur** [xaur/decred-news#106]
DRAFT: https://github.com/xaur/decred-news/blob/draft10/journal/201910.md

**Dependencies**: Send notice in advance so people can plan their work.

- [x] Title image via an [issue in dcrdesign](https://github.com/decred/dcrdesign/issues) (notify by: 20th, ready by: 3rd)
  * GitHub version must be 768x384 px
  * ask for image title and description (saender)
- [x] first time contributors (degeri or s_ben)
- [x] dev activity stats (degeri or s_ben)
- [x] Ditto update (cryptoleslie or liz_bagot, notify by: 28th, ready by: 3rd)
- [x] Outreach update (Dustorf, notify by: 28th, ready by: 3rd)

**End of month snapshots**: Snapshot data that is hard to obtain later.

- [x] snapshot social media stats
  * Politeia, Matrix, Discord, Slack, Telegram, Twitter, Reddit, YouTube, Facebook, LinkedIn, GitHub dcrd, Instagram
  * (opt) non-English platforms: Telegram (es pt zh it ru), Twitter (es), Instagram
- [x] staking
  * 30-day average ticket price (https://dcrstats.com)
- [x] nodes
  * node count, versions, etc (ask chappjc or #dev or dcr.farm)

**Release**

- [x] Extract more dev updates from GitHub repos
- [x] Complete Development (by: 2nd)
- [x] Complete People (by: 2nd)
- [x] PO review Development and People
- [x] Ask in #dev to review Development and People
- [x] Complete Events
- [x] (opt) Ask #event_planning to review Events
- [x] Complete Outreach
- [x] Complete Media
- [x] (opt) Ask #marketing to review Outreach+Events+Media
- [x] Complete Governance
- [x] Complete Network
- [x] Complete Mining, Integrations, Adoption
- [x] (opt) Find interesting Reddit and Twitter threads
- [x] Complete Community Discussions
- [x] Complete Markets
- [x] Complete Relevant External
- [x] (opt) Ask quoted people to review their texts
- [x] Complete Intro
- [x] Pre-review: check spelling, grammar, test that links work
- [x] PO review of all sections
- [x] Add [title image](https://github.com/xaur/decred-news/blob/docs/guidelines.md#title-image)
- [x] Ask #writers_room to review (by: 3rd)
- [x] Incorporate all feedback
- [x] Resolve all [TODOs](https://github.com/xaur/decred-news/blob/docs/guidelines.md#todos)
- [x] Carefully check that all people who contributed or gave feedback are in [credits](https://github.com/xaur/decred-news/blob/docs/guidelines.md#how-to-give-credit)
- [x] [Lint](https://github.com/xaur/decred-news/blob/docs/guidelines.md#linting)
- [x] Remove `{DRAFT}` from title
- [x] Pre-release review
- [x] Publish on GitHub with a single commit to `gh-pages`
- [x] Publish on Medium (richardred or Haon)
- [x] (opt) Verify Medium version, check for formatting issues
- [x] Choose the best link to disseminate: Medium vs GitHub Pages vs decredcommunity, etc
- [x] Submit best link to r/decred
- [x] Tweet best link via decredproject. Don't forget to add title image. (by: 6th)
- [x] If bugs are found in the published version quickly after the release, add commits to the draft branch and amend the commit on `gh-pages`
- [x] Fast-forward `master` to `gh-pages`
- [x] If changes are necessary after the release (>1 h), add extra commits to both `master` and `gh-pages`.
- [x] [Update index](https://github.com/xaur/decred-news/blob/docs/guidelines.md#updating-index)
- [x] Create Git tag from the draft branch, e.g. `archive/draft1901`
- [x] Delete the draft branch
- [x] (opt) Update `journal-template.md` in `docs` branch
- [x] (opt) Update this release checklist template in the `docs` branch

**Comment by xaur** [comment]
Image issue: https://github.com/decred/dcrdesign/issues/177

**Comment by nottrunner** [comment]
Akin Sawyerr updates 

Oct 29 - World Crypto Conference - Las Vegas, USA. @akinsawyerr represented Decred on the Governance in Practice panel, and gave a presentation called Governing the Crypto commmons where he outlined Decred's  governance process and shared insights derived from a year's worth of Politeia data. Akin also sat for several interviews and meetings arranged by Ditto.

**Comment by nottrunner** [comment]
Oct 23 - World Bank Innovation Lab - Washington D.C. @akinsawyerr conducted an overview presentation on Decred to staff members of the World Bank Innovation Lab. The conversation also delved into ways the World Bank could leverage Decred infrastructure in some of its proof of concept projects.

**Comment by xaur** [comment]
Thanks @nottrunner, good stuff. Added.

**Comment by luisantoniocrag** [comment]
event report of : Oct 17 - Technological University - Morelia, Mexico

https://github.com/decredcommunity/events/pull/22/commits/6a521b84d634e499cbfed73c9c532fd0106e48f9?short_path=b3efd98#diff-b3efd98aa63ab3bb09cb87966f499631

**Comment by luisantoniocrag** [comment]
Oct 24 - Cryptocurrency Workshop 
please add =)

link of event: https://twitter.com/Decred_ES/status/1187206051000655872

add to description:
The workshop, unlike the October 17 talk, was much more detailed, and although it was 15 people, all of them thoroughly understood the operation of Blockchain and Decred. Not only on the technical side but also financially.

**Comment by xaur** [comment]
Thanks @luisantoniocrag! Oct 24 added, will link Oct 17 report once it is merged.

**Comment by xaur** [comment]
### Dev

#### dcrd

> updating the block header commitments has been implemented

this is new thing, they were added, not updated

> Major v1.5 release has landed on master.

It is release _candidate_ and it has not really "landed" on master. We say this about big changes that were long in development and were finally merged to master. In case of this RC, a release branch was branched off master and got a single commit added that sets the dependencies for release builds.

> Updates to various modules has begun, including for the rpcclient, stake, txscript, blockchain. mempool and connmgr modules.

Most of the linked PRs just bump version numbers to start new dev cycle and don't carry any real updates. Dropping.

Good work cutting the v1.5 release notes. It is still a big chunk of text but it is packed with feature highlights, so well deserves its space.

#### dcrwallet

> Bug fixes, performance improvements and updates to modules including secp256k1, backport, errors, among others.

I was confused at first because there was no info in the linked pull requests. But I think the general "bug fixes and performance improvements" refers to all small pull requests that don't receive individual reporting.

These are good but too small for individual reporting. Group in "Bug fixes":

- Undeprecate select JSON-RPC methods and fields: `https://github.com/decred/dcrwallet/pull/1562`
- Partition watched outpoints into multiple LoadTxFilter calls: `https://github.com/decred/dcrwallet/pull/1564`
- Fix out of bounds slice access : `https://github.com/decred/dcrwallet/pull/1566`
- Serialize requests to load transaction filter with addrs: `https://github.com/decred/dcrwallet/pull/1585`
- Add --nodcrdproxy flag to never dial dcrd through a proxy.: `https://github.com/decred/dcrwallet/pull/1589`
- Never use proxy when dialing localhost: `https://github.com/decred/dcrwallet/pull/1590`
- Associate wallet with spv syncer during Run: `https://github.com/decred/dcrwallet/pull/1591`
- Derive and load watched addresses outside of DB view.: `https://github.com/decred/dcrwallet/pull/1592`
- Do not watch future addresses when transactions are not fully synced.: `https://github.com/decred/dcrwallet/pull/1593`
- Derive mixed output address after dialing CSPP server: `https://github.com/decred/dcrwallet/pull/1594`
- Prevent errgroup teardown from max proxy connections: `https://github.com/decred/dcrwallet/pull/1596`
- Do not cleanly unload wallet during panic: `https://github.com/decred/dcrwallet/pull/1597`
- Combine logic for loading and watching future addresses: `https://github.com/decred/dcrwallet/pull/1598`
- Throttle goroutines spawned by MixAccount: `https://github.com/decred/dcrwallet/pull/1600`
- Ensure pooled connection is always closed: `https://github.com/decred/dcrwallet/pull/1601`
- Disable RPC client pong deadline: `https://github.com/decred/dcrwallet/pull/1602`
- Avoid OOB panic when subslicing outpoints to mix: `https://github.com/decred/dcrwallet/pull/1603`

Let's group this as "Trace code added to get more profiling data":

- Add trace tasks for JSON-RPC calls and notifications: `https://github.com/decred/dcrwallet/pull/1567`
- Add a trace region for all db views and updates: `https://github.com/decred/dcrwallet/pull/1568`
- Add another trace region once a DB tx has been opened : `https://github.com/decred/dcrwallet/pull/1569`

Group as "Code updated to use new modules from dcrwallet and latest modules from dcrd.":

- Use released errors/v2 module from validate: `https://github.com/decred/dcrwallet/pull/1576`
- Use released modules and latest dcrd submodules: `https://github.com/decred/dcrwallet/pull/1577`
- Use released dcrd client module from wallet: `https://github.com/decred/dcrwallet/pull/1579`

This is "Merged some work to validate new block layout post v1.5" (although this didn't make it into the final release):

- Add validation func for combined merkle root: `https://github.com/decred/dcrwallet/pull/1575`
- Allow DCP0005 merkle roots to successfully validate: `https://github.com/decred/dcrwallet/pull/1580`

Internal dependency management churn, dropping:

- Prepare chain/v3.0.0, spv/v3.0.0: `https://github.com/decred/dcrwallet/pull/1581`
- Prepare ticketbuyer/v4.0.0: `https://github.com/decred/dcrwallet/pull/1582`
- Branch off for 1.5 release: `https://github.com/decred/dcrwallet/pull/1583`
- Update version module to report as v1.5.0: `https://github.com/decred/dcrwallet/pull/1584`
- Update to secp256k1/v2 module: `https://github.com/decred/dcrwallet/pull/1586`
- Tidy the wallet go.{mod,sum}: `https://github.com/decred/dcrwallet/pull/1587`
- Update submodules and backport main module improvements: `https://github.com/decred/dcrwallet/pull/1604`
- Update pgpwordlist and walletseed modules: `https://github.com/decred/dcrwallet/pull/1605`

This is part of CSPP work, but this just adds command definitions and does nothing else. This must be reported together with bigger CSPP work:

- Define importxpub, mixaccount, and mixoutput command types : `https://github.com/decred/dcrwallet/pull/1572`

Missed:

- initial CoinShuffle++ `https://github.com/decred/dcrwallet/pull/1541`
- import arbitrary xpub accounts `https://github.com/decred/dcrwallet/pull/1471`

These have been long in dev and we covered them quite a few times, but merge in master is still notable.

#### Decrediton

These were already covered in the composed text, please don't duplicate them:

- [helpers] Add missing agendas: `https://github.com/decred/decrediton/pull/2219`
- Add warn when upgrading to new dcrd database version: `https://github.com/decred/decrediton/pull/2228`
- Responsive views: Help - Sources/Tutorials/Logs: `https://github.com/decred/decrediton/pull/2202`

Covered these collectively as "bug fixes":

- private passphrase spacing `https://github.com/decred/decrediton/issues/2201`
- Set artifactName: `https://github.com/decred/decrediton/pull/2195`
- Fix recent transactions icons: `https://github.com/decred/decrediton/pull/2206`
- Fix cancel rescan button alignment: `https://github.com/decred/decrediton/pull/2207`
- [ln] Add syncedToChain check at connect: `https://github.com/decred/decrediton/pull/2218`
- [sidebar] Have sidebar leave bottom more aggresively: `https://github.com/decred/decrediton/pull/2221`
- [Multi] Fix views tweaks: `https://github.com/decred/decrediton/pull/2229`
- Testnet Logo in Launcher: `https://github.com/decred/decrediton/pull/2230`
- Fix AccountsSelect on SendTab: `https://github.com/decred/decrediton/pull/2240`
- Fix input field size and alignment on `Seed Clipboard Copy Warning` modal: `https://github.com/decred/decrediton/pull/2241`
- Show missing log headers on smaller screens: `https://github.com/decred/decrediton/pull/2242`
- [maindev] Wait for dcrd's rpc.cert to be generated: `https://github.com/decred/decrediton/pull/2243`

These are covered by "responsive design" work:

- [tickets] fix purchase tickets tweaks: `https://github.com/decred/decrediton/pull/2205`

By the way, one way to link multiple related PRs collectively is by using a GitHub search query:

https://github.com/decred/decrediton/pulls?q=is%3Apr+merged%3A2019-10-01..2019-10-31+responsive

This works only if all PRs have the necessary keyword in them.

Maintenance:

- [deps] Update electron-builder and update package scripts: `https://github.com/decred/decrediton/pull/2226`
- build: build and test with latest nodejs: `https://github.com/decred/decrediton/pull/2236`

#### Politeia

Missed:

- `politeiavoter` was made more resilient to network errors `https://github.com/decred/politeia/pull/1022`.

#### dcrstakepool

> All in-line JavaScript has been removed `https://github.com/decred/dcrstakepool/pull/560`

Love stuff like this!

These were already covered in the composed text:

- multi: Update modules: `https://github.com/decred/dcrstakepool/pull/554`

Group as "refactoring" or "cleanup":

- multi: drop udb usage: `https://github.com/decred/dcrstakepool/pull/567`
- Move page specific code out of global.js: `https://github.com/decred/dcrstakepool/pull/561`
- params: Remove unused WalletRPCServerPort: `https://github.com/decred/dcrstakepool/pull/553`
- stakepoold: Move rpc/rpcserver to rpc/server: `https://github.com/decred/dcrstakepool/pull/550`

Group as "maintenance":

- build: upgrade golangci-lint: `https://github.com/decred/dcrstakepool/pull/570`
- modules: sync to latest: `https://github.com/decred/dcrstakepool/pull/565`
- bump version to 1.5.0-pre: `https://github.com/decred/dcrstakepool/pull/558`

Group as "bug fixes":

- Prevent header links from moving: `https://github.com/decred/dcrstakepool/pull/559`

Not merged:

- Improve golint score: `https://github.com/decred/dcrstakepool/issues/564`
  - not even a PR

#### dcrlnd

Already covered in text:

- Allow use with a remote wallet `https://github.com/decred/dcrlnd/pull/40`

Maintenance, group as "cleanup":

- Update dcrd/dcrwallet packages: `https://github.com/decred/dcrlnd/pull/39`

Group as "minor fixes":

- chainntnfs: Do not require txindex in dcrd `https://github.com/decred/dcrlnd/pull/41`
- Nodelay keyfam wallet `https://github.com/decred/dcrlnd/pull/44`
- build: Update to 0.2.0 `https://github.com/decred/dcrlnd/pull/45`
- build: replace travis-ci with ci via github actions `https://github.com/decred/dcrlnd/pull/47`
- build: set GOPATH for tests `https://github.com/decred/dcrlnd/pull/48`
- Fix GitHub actions `https://github.com/decred/dcrlnd/pull/49`
- Remove package replacements `https://github.com/decred/dcrlnd/pull/50`
- Fixed broken link to docker docs `https://github.com/decred/dcrlnd/pull/51`

#### dcrandroid

Text looks almost identical to September. Upon checking I see that some work was done to advance those two huge tasks, but the text may appear like nothing changed. It is normal to work on big features for several months but I recommend to zoom in a bit to see what has advanced since last month.

#### dcrios

> New functions added to query transactions with filters for type and direction. History page has been overhauled to incorporate new filters.

This is identical to September, but unlike dcrandroid, these changes were finished and merged in Sep. Dropping.

### Dev recap

The oneline per-project summarries for Decrediton, dcrlnd and dcrstakepool were almost perfect! Only the last of them "survived" till the release because the other ones were superseded by the coverage for release candidates. But it is getting better.

@s-ben I suggest to stop adding lists of unprocessed PRs to the draft. It will save time for you and for me, because I check the GitHub query for PRs [merged in that month](https://github.com/decred/dcrstakepool/pulls?q=is%3Apr+merged%3A2019-10-01..2019-10-31) anyway.

Also, please let me know if my detailed per-PR notes above are helpful and whether I should keep making them.

**Comment by xaur** [comment]
### Governance

> For more in depth coverage of governance see Politeia Digest issues [23](https://www.blockcommons.red/politeia-digest/issue023/) and [24](https://www.blockcommons.red/politeia-digest/issue024/).

In case anyone is wondering, I'm sligthly changing link text to be easier to click per [Guidelines/Links](https://github.com/xaur/decred-news/blob/docs/guidelines.md#link-text):

> [issue 23](https://www.blockcommons.red/politeia-digest/issue023/) and [issue 24](https://www.blockcommons.red/politeia-digest/issue024/).

**Comment by xaur** [comment]
### Network

> \- {Monthly hashrate low around 370 Ph/s correlates with drop in DCR price}  
> \+ The drop in prices correlated with the hash rate dropping momentarily before it recovered again.

Sorry I didn't put it clearly in the note, but the goal here was to show a negative change in a critical network health parameter. We don't try to paint always positive picture. As of Nov 4, the 30 day change is `-22.9%` per [dcrdata](https://explorer.dcrdata.org/). If something is going bad we want to put it loud and clear so that stakeholders can take action.

**Comment by xaur** [comment]
### General

@s-ben I noticed your messages in Git commits but they are very easy to miss. Please put all messages you'd like the group to read in our chat room or in the associated pull request.

> Got sign-off from raedah for dcrandroid and dcrios

Please add people to credits as soon as you get something from them, so that we don't miss anyone.

---

Done with my nitpicks, the release was epic! Thank you all. I'll get myself a small cake to celebrate and recommend the same for you guys.
